compiler/semobjconstr.nim

```diff
@@ -392,17 +392,19 @@ proc semObjConstr(c: PContext, n: PNode, flags: TExprFlags): PNode =
       result.typ.flags.incl tfHasOwned
   if t.kind != tyObject:
     localError(c.config, n.info, errGenerated, "object constructor needs an object type")
-    return
+    return errorNode(c, result)
 
   # Check if the object is fully initialized by recursively testing each
   # field (if this is a case object, initialized fields in two different
   # branches will be reported as an error):
   var constrCtx = initConstrContext(t, result)
   let initResult = semConstructTypeAux(c, constrCtx, flags)
+  var hasError = false
 
   # It's possible that the object was not fully initialized while
   # specifying a .requiresInit. pragma:
   if constrCtx.missingFields.len > 0:
+    hasError = true
```

Comment:
why not: `return errorNode(c, result)` ?

likewise in other cases below where you have `hasError = true`

Reply:
Because then the analysis changes and doesn't work in cases where you wish to report all errors, which is in fact safe and better to do so.

Reply:
Added a small comment to explain hasError purpose.

Reply:
the flip side of the argument is redundant/bubbling errors, refs https://github.com/nim-lang/Nim/issues/16178, making nim check less useful; ideally I want all errors reported by nim check/nim r + --errormax:N to be unrelated (and non-causal)

as the example in tests/objects/t17437.nim shows, there are indeed redundant errors being reported

Reply:
The redundant messages is a different issue.

So if there is an object with two or more fields, and you attempt to construct it but the initialization of two fields is a mismatch in type, then we should report each of those (wait to the end of the loop). Which is what it was doing before and that behavior makes sense, otherwise you'll get under reporting.